### PR TITLE
TRITON-1961 add Jenkinsfile to sdc-headnode

### PR DESCRIPTION
**Pull request by timfoster**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr: '30'))
+        timestamps()
+    }
+    agent none
+
+    // build once per day
+    triggers {
+        cron('H H * * *')
```
Comment (trentm) [inline Jenkinsfile]: This is the same as currently, which is all good. I wonder if there is value in scheduling this to complete a little before nightly reflashes kick off? Which are:

- nightly-1, 2, 3:

```
# Early in the morning PDT
H 4 * * *
```

- nightly-4:

```
# Early in the morning PDT
H 6 * * *
```

Current builds are ~25 minutes (https://jenkins.joyent.us/job/headnode/buildTimeTrend), so setting this to something like:

```
// A couple hours before nightly reflashes tend to kick off, at 'H 4 * * *',
// so we get reasonably up to date headnode images. 
cron('H 2 * * *')
```


Thoughts?  Could be a separate change as well.

Comment (timfoster) [reply Jenkinsfile]: Yep, that's sensible - I'll fold that into this change, thanks.

**Comment by timfoster** [comment]
Sorry, we're hitting the build failure fixed by TOOLS-2298 here, so I need to bump deps/eng to bring that in

```
[root@950abac8 ~/data/jenkins/workspace/headnode-PR-33-check]# ./deps/eng/tools/check-copyright
error: buildtools/lib/common.sh: copyright year not updated to 2020: '# Copyright 2019 Joyent, Inc.'
[root@950abac8 ~/data/jenkins/workspace/headnode-PR-33-check]#
```

**Comment by timfoster** [comment]
huh, no it's something else, we've already got that fix. Will investigate.

**Comment by timfoster** [comment]
oh, I see - I just need to merge master into my branch. That was a bit confusing.

```
diff --git a/buildtools/lib/common.sh b/buildtools/lib/common.sh
index 740f011..72a55a2 100755
--- a/buildtools/lib/common.sh
+++ b/buildtools/lib/common.sh
@@ -6,11 +6,9 @@
 #

 #
-# Copyright 2020 Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #

-TOP="$( cd "$(dirname "${BASH_SOURCE[0]}")/../../" >/dev/null 2>&1 && pwd )"
-
 PLATFORM=$(uname -s)

 TAR="tar"
@@ -149,5 +147,5 @@ function unmount_root_image
 function build_spec
 {
     local thing=$1;
-    ${JSON} -f "${TOP}/build.spec.merged" ${thing}
+    ${JSON} -f build.spec.merged ${thing}
 }
```

**Pull request merged**